File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int r,temp, sum=0;   
        int n = x;
        while (n>0){
            r = n%10;
            sum = sum*10 + r;
            n = n/10;
        }
        return sum == x;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
import math

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0:
            return False
        else:
            arr = [char for char in str(x)]
            if len(arr) == 1:
                return True
            for i in range(len(arr)):
                if arr[i] != arr[-i-1]:
                    return False
                if i+1 == math.ceil(len(arr) / 2):
                    return True
                    
    
        